client/widgets/CArtifactHolder.cpp

```diff
@@ -260,18 +260,29 @@ bool ArtifactUtilsClient::askToAssemble(const CGHeroInstance * hero, const Artif
 	assert(hero);
 	const auto art = hero->getArt(slot);
 	assert(art);
-	auto assemblyPossibilities = ArtifactUtils::assemblyPossibilities(hero, art->getTypeId(), ArtifactUtils::isSlotEquipment(slot));
 
-	for(const auto combinedArt : assemblyPossibilities)
+	if(hero->tempOwner == LOCPLINT->playerID)
```

Comment:
Will this work correctly (for example) 
- if my hero dies in attack on AI and AI gets an option to assemble an artifact
- I AI attacks me, I defeat AI hero and get an option to assemble artifact? (during AI turn)